Core/Inc/main.h

```diff
@@ -36,7 +36,7 @@ extern "C" {
 
 /* Exported types ------------------------------------------------------------*/
 /* USER CODE BEGIN ET */
-
+typedef enum {Saved = 0, Transmitted = 1} USER_DATA_EVENTS;
```

Comment:
```suggestion
typedef enum {SAVED = 0, TRANSMITTED = 1} USER_DATA_EVENTS_t;
```
"_t" at the end means type, better to use with typedef keyword 
Constants, as enum, better to write in upper case